src/app/share/views/SharedLinksView/SharedView.tsx

```diff
@@ -362,6 +397,72 @@ export default function SharedView(): JSX.Element {
     await moveItemsToTrash([itemToTrash]);
   };
 
+  const onUploadFileButtonClicked = (): void => {
+    errorService.addBreadcrumb({
+      level: 'info',
+      category: 'button',
+      message: 'File upload button clicked',
+      data: {
+        currentFolderId: currentFolderId,
+      },
+    });
+    fileInputRef.current?.click();
+  };
+
+  const onUploadFileInputChanged = async (e) => {
+    const files = e.target.files;
+    dispatch(
+      storageActions.setItems({
+        folderId: currentParentFolderId as number,
+        items: shareItems as unknown as DriveItemData[],
+      }),
+    );
+
+    if (files.length < 1000 && currentParentFolderId) {
```

Comment:
Invert the if and return in that case. Thus, you can remove some nesting/code complexity